api/v1alpha1/oadp_types.go

```diff
@@ -264,3 +271,41 @@ func (dpa *DataProtectionApplication) BackupImages() bool {
 func init() {
 	SchemeBuilder.Register(&DataProtectionApplication{}, &DataProtectionApplicationList{}, &CloudStorage{}, &CloudStorageList{})
 }
+
+// AutoCorrect is a collection of auto-correction functions for the DPA CR
+// These auto corrects are in-memory only and do not persist to the CR
+// There should not be another place where these auto-corrects are done
+func (dpa *DataProtectionApplication) AutoCorrect() {
+	//check if CSI plugin is added in spec
+	if hasCSIPlugin(dpa.Spec.Configuration.Velero.DefaultPlugins) {
```

Comment:
Instead of removing possible duplicates, could there also be a check here for the appended value existing in dpa?

Reply:
Removing duplicate is computationally expensive.. rather do it once.. I think line 299-300 covers this so perhaps no need to use unique function here and just append.

Reply:
Minimize number of for loops against the array

Reply:
checking for existing is also another for loop.. so

Reply:
WDYT? https://github.com/openshift/oadp-operator/pull/987/commits/6c3b573aba87f0e7f0f456091d50e1224f9e67a1